ui/dashboards/src/context/DashboardProvider.tsx

```diff
@@ -49,33 +52,56 @@ export function usePanels() {
 }
 
 export function useLayouts() {
-  const { layouts, setLayouts } = useStore(({ layouts, setLayouts }) => ({ layouts, setLayouts }));
-  return { layouts, setLayouts };
+  const { layouts, setLayouts, addLayout, addItemToLayout } = useStore(
+    ({ layouts, setLayouts, addLayout, addItemToLayout }) => ({
+      layouts,
+      setLayouts,
+      addLayout,
+      addItemToLayout,
+    })
+  );
+  return { layouts, setLayouts, addLayout, addItemToLayout };
 }
 
 export function useEditMode() {
   const { isEditMode, setEditMode } = useStore(({ isEditMode, setEditMode }) => ({ isEditMode, setEditMode }));
   return { isEditMode, setEditMode };
 }
 
+export function useDashboard() {
+  const getDashboard = useStore((state) => state.getDashboard);
+  return { getDashboard };
+}
+
 export function DashboardProvider(props: DashboardProviderProps) {
   const {
     children,
-    initialState: {
-      dashboardSpec: { layouts, panels },
-      isEditMode,
-    },
+    initialState: { dashboardSpec, isEditMode },
   } = props;
 
+  const { layouts, panels } = dashboardSpec;
+
   return (
     <Provider
       createStore={() =>
-        create((set) => ({
+        create((set, get) => ({
           layouts,
           panels,
           isEditMode: !!isEditMode,
           setEditMode: (isEditMode: boolean) => set({ isEditMode }),
           setLayouts: (layouts: LayoutDefinition[]) => set({ layouts }),
```

Comment:
I wonder if we'll even need setLayouts, if we'll just use addLayout / removeLayout type of functions instead 🤔 

We can wait and see how it turns out, and you might have a plan for it. Mostly just noting to myself that maybe we'll end up not needing this.

Reply:
my original thinking is that we'll have to update the whole layout when user repositions the panels in a grid, but that's only one layout, so we probably have to refactor this :)